allennlp/training/trainer.py

```diff
@@ -673,6 +676,23 @@ def train(self) -> None:
                 formatted_time = time.strftime("%H:%M:%S", time.gmtime(estimated_time_remaining))
                 logger.info("Estimated training time remaining: %s", formatted_time)
 
+            epochs_trained += 1
+
+        training_elapsed_time = time.time() - training_start_time
+        metrics = {
+                "training_duration": time.strftime("%H:%M:%S", time.gmtime(training_elapsed_time)),
+                "training_start_epoch": epoch_counter,
+                "training_epochs": epochs_trained
```

Comment:
additionally, why "training_epochs" here rather than "training_stop_epoch"?

Reply:
@joelgrus I want a name for the number of epochs trained in the loop.  We might resume training at epoch 2 and end on epoch 5.  In that case `trained_epochs` would be 3.

At the last, I'll add a comment.  Do you have an idea for a better name?

Reply:
no, the name is fine, I just meant that if you have `training_start_epoch` and `training_stop_epoch` (and you do), then `training_epochs` is superfluous

Reply:
Yep--sorry misunderstood.

Reply:
@joelgrus counting explicitly seems like the easiest and most foolproof way to get this count.  Otherwise I need to handle the case when the for loop is empty.

Normally the number of epochs computed is `epoch - epoch_counter + 1`, but if the range is empty then it's 0.